portfolio/src/main/webapp/script.js

```diff
@@ -56,7 +57,84 @@ function getRandomFile(element, path, fileName, numElems) {
  * @param {DOM element} element - an element in the DOM for selecting
  */
 function fetchAndModify(path, element) {
-  fetch(path).then(response => response.text()).then(content => 
-    element.innerHTML= content
-  )
+  fetch(path)
+      .then(response => response.text())
+      .then(content => element.innerHTML = content)
 }
+/*
+ * This code creates an innerHTML and then reads its text to remove special
+ * characters from the input
+ * @param {string} str The input to remove special characters from to prevent
+ *     scripting attacks
+ */
+var sanitizeHTML = function(str) {
+  var temp = document.createElement('div');
+  temp.textContent = str;
+  return temp.innerHTML;
+};
+
+
+/*
+ * This function fetches the value return by the given data service and then
+ * modifies the content of a div to store each of the response values as a
+ * paragraph.
+ * @param {string} path The url of the data service to use
+ */
+function fetchAndChangeBody(path) {
+  fetch(path).then(response => response.json()).then(json => {
+    let index = 0;
+    for (index in json) {
```

Comment:
I think `json` is actually a set of key-value pairs, right? I'd recommend renaming `index` (usually only used for integers) to `key`.

Reply:
The json generated in this request is actually just a list so of the form [element1, element2, element3] and thus doesn't actually have a start key. That's why I used index in this case

Reply:
Ah, I see, thanks. In that case, prefer the `for`-`of` form: https://google.github.io/styleguide/jsguide.html#features-for-loops

Reply:
You can rename  json to jsonArray or contentList or whatever to make it more clear.